### PR TITLE
Finish Responsive TTD with stubbed clients and store schemas

### PR DESCRIPTION
**Pull request by ableegoldman**
Fills out the TopologyTestDriver ecosystem of stubbed clients, schemas, writers, and stores with a basic in-memory map for use in a TTD test app

### CODE REVIEW

File: kafka-client/src/main/java/dev/responsive/kafka/store/ResponsivePartitionedStore.java [hunk 1]
```diff
@@ -234,17 +234,15 @@ public Position getPosition() {
...
 
   @Override
   public void close() {
+    // no need to flush the buffer here, will happen through the kafka client commit as usual
     if (storeRegistry != null) {
       storeRegistry.deregisterStore(registration);
     }
-    // the client is shared across state stores, so only the
-    // buffer needs to be flushed
-    flush();
```
Comment (ableegoldman) [inline kafka-client/src/main/java/dev/responsive/kafka/store/ResponsivePartitionedStore.java:248]: just some minor cleanup, the comment and `flush` call are out of date (the flush invoked here is now a no-op)

File: responsive-test-utils/src/main/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriver.java [hunk 2]
```diff
@@ -81,12 +103,26 @@ private static Properties testDriverProps(
...
-      super(new ResponsiveConfig(new HashMap<>()));
+  private static class TTDMockAdmin extends MockAdminClient {
+    private static final Node BROKER = new Node(0, "dummyHost-1", 1234);
+    public TTDMockAdmin() {
+      super(Collections.singletonList(BROKER), BROKER);
     }
 
-    // TODO: override CassandraClient methods to stash data in a local in-memory stub
+    @Override
+    public DescribeTopicsResult describeTopics(Collection<String> topicNames) {
```
Comment (agavra) [inline responsive-test-utils/src/main/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriver.java:113]: are we dynamically creating topics on each `describeTopics` call? why is that necessary?

File: responsive-test-utils/src/main/java/dev/responsive/kafka/api/CassandraClientStub.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  public CassandraClientStub(final Properties props) {
+    super(new ResponsiveConfig(props));
+    kvSchema = new TTDKeyValueSchema(this);
+    windowedSchema = new TTDWindowedSchema(this);
+  }
+
+  @Override
+  public ResultSet execute(final Statement<?> statement) {
+    return null;
```
Comment (agavra) [inline responsive-test-utils/src/main/java/dev/responsive/kafka/api/CassandraClientStub.java:48]: throughout this PR there's a pattern of returning `null` for all Cassandra related APIs, I'm wondering if it might make sense to instead return stubs that just throw exceptions if any of their methods are used? we can save that for a follow-up if this ever causes issues, but it makes me a bit nervous 😅

**Review by agavra — APPROVED** [review]
very nice, super quick turnaround 🐎 thanks @ableegoldman

**Review by apurvam — APPROVED** [review]
Thanks for the quick turnaround! LGTM!

File: responsive-test-utils/src/main/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriver.java (hunk 2 above)
Comment (ableegoldman) [reply responsive-test-utils/src/main/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriver.java:113]: Hahaha yeah I know, this feels very silly -- but if the topic doesn't exist then we get an `UnknownTopic` error on the `describeTopics` call in `ResponsiveConfig#getSubPartitioner`, and this was the quickest/easiest way to fix that. It didn't seem worth going beyond a hacky solution at this point

File: responsive-test-utils/src/main/java/dev/responsive/kafka/api/CassandraClientStub.java (hunk 3 above)
Comment (ableegoldman) [reply responsive-test-utils/src/main/java/dev/responsive/kafka/api/CassandraClientStub.java:49]: That's fair, this was another instance of the philosophy of not going above and beyond unnecessarily at this point but I completely agree -- I put in a TODO but feel free to file a ticket

File: responsive-test-utils/src/test/java/dev/responsive/kafka/api/TopologyTestDriverExampleTest.java [hunk 4]
```diff
@@ -31,19 +31,19 @@
...
   public void shouldRunWithoutResponsiveConnection() {
     // Given:
     final Properties props = new Properties();
     props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
     props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
 
-    final Topology topo = topology();
-    final TopologyTestDriver driver = new ResponsiveTopologyTestDriver(topo, props);
+    final Topology topology = topology();
+    final TopologyTestDriver driver = new ResponsiveTopologyTestDriver(topology, props);
```
[the remark this answers is not in the document]
Comment (ableegoldman) [reply responsive-test-utils/src/test/java/dev/responsive/kafka/api/TopologyTestDriverExampleTest.java:46]: Well I personally know that it's using the Responsive stuff just from debugging the CassandraClient/Schema/Writer stubs lol -- but I know what you mean.

This might best be achieved by checking the return values once we have those stubbed instead of returning null everywhere as this PR currently does. Anyways I'm going to do the ttl + expanded test coverage in an (immediate) followup PR so I'll see what we can do there
